src/test/kotlin/com/example/passmanager/repositories/PassRepositoryImplTest.kt

```diff
@@ -11,207 +10,249 @@ import org.assertj.core.api.Assertions.assertThat
 import org.bson.types.ObjectId
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.boot.test.context.SpringBootTest
-import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate
+import org.springframework.data.mongodb.core.aggregation.Fields
 import org.springframework.data.mongodb.core.exists
+import org.springframework.data.mongodb.core.findById
 import org.springframework.data.mongodb.core.query.Criteria.where
 import org.springframework.data.mongodb.core.query.Query.query
 import org.springframework.data.mongodb.core.query.isEqualTo
+import reactor.kotlin.test.test
 import java.math.BigDecimal
 import java.time.LocalDate
 import java.time.ZoneId
 import kotlin.test.Test
-import kotlin.test.assertEquals
 import kotlin.test.assertFalse
-import kotlin.test.assertTrue
 
 @SpringBootTest
 @WithMongoTestContainer
-class PassRepositoryImplTest {
+internal class PassRepositoryImplTest {
 
     @Autowired
-    private lateinit var mongoTemplate: MongoTemplate
+    private lateinit var mongoTemplate: ReactiveMongoTemplate
 
     @Autowired
     private lateinit var passRepository: PassRepository
 
     @Test
     fun `finding pass by existing id should return pass by id`() {
         // GIVEN
-        val inserted = mongoTemplate.insert(passToCreate)
+        val insertedPass = mongoTemplate.insert(passToCreate).block()
 
         // WHEN
-        val passById = passRepository.findById(inserted.id.toString())
+        val passById = passRepository.findById(insertedPass!!.id.toString())
 
         // THEN
-        assertThat(passById?.id).isNotNull().isEqualTo(inserted.id)
+        passById.test()
+            .assertNext {
+                assertThat(it).usingRecursiveComparison()
+                    .ignoringFields(MongoPass::purchasedAt.name)
```

Comment:
why ignoring this field?

Reply:
Problem is somehow Instant fields doesn't have same precision. Example:

`Expected :MongoPass(id=6705200a54bbb3052f03efbb, purchasedFor=10, passOwnerId=null, passTypeId=null, purchasedAt=2024-10-08T12:05:30.473292Z, version=1)

Actual   :MongoPass(id=6705200a54bbb3052f03efbb, purchasedFor=10, passOwnerId=null, passTypeId=null, purchasedAt=2024-10-08T12:05:30.473Z, version=1)`

So I had to ignore Instant fields :/